src/Workspaces/Remote/ServiceHub/Shared/ServiceHubServiceBase.cs

```diff
@@ -27,8 +26,22 @@ internal abstract class ServiceHubServiceBase : IDisposable
         protected readonly AssetStorage AssetStorage;
         protected readonly CancellationToken CancellationToken;
 
+        /// <summary>
+        /// Session Id of this service. caller and callee share this id which one
+        /// can use to find matching caller and callee when debugging or logging
+        /// </summary>
         private int _sessionId;
+
+        /// <summary>
+        /// Mark whether the solution checksum it got is for primary branch or not
```

Comment:
we need a comment somewhere to explain why this is impoirtant.  why do we care if it's from the primary branch or not?  How do we treat things differently if it's from the primary branch or not.